src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts

```diff
@@ -168,27 +168,54 @@ async function webviewPreloads(ctx: PreloadContext) {
 		readonly workspace: { readonly isTrusted: boolean };
 	}
 
-	interface ScriptModule {
-		activate(ctx?: RendererContext): Promise<RendererApi | undefined | any> | RendererApi | undefined | any;
+	interface RendererModule {
+		activate(ctx: RendererContext): Promise<RendererApi | undefined | any> | RendererApi | undefined | any;
+	}
+
+	interface KernelPreloadContext {
+		getState(): { [key: string]: unknown; };
+		setState(data: { [key: string]: unknown; }): void;
+		postMessage(msg: unknown): void;
```

Comment:
I don't think this should be exposed

Reply:
Should we expose `acquireVsCodeApi` on this object instead? We currently pass the global `acquireVsCodeApi` function as a global to preload scripts: https://github.com/microsoft/vscode/blob/742a67937e7fdee0b35bb711f73d2fff7dccb1d0/src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts#L498

I'd agree that this seems fishy but wasn't sure if some preload scripts were relying on it

Reply:
I don't think there's a reason extensions would want to use the "raw" post message. In order to do anything useful they would need to wrap their messages up and send things compatible with the webview<->backlayerwebview protocol, which I don't think we want to allow much less make API.